### ISSUE TITLE
Reindex Issue

### ISSUE THREAD

**Issue by an2riy** [Blarc/gitlab-template-lint-plugin#127]
Hello. 
Each time when I close and open the project the reindexation is triggered.
```
PhpStorm 2023.1.2
Build #PS-231.9011.38, built on May 17, 2023
Runtime version: 17.0.6+10-b829.9 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.15.0-73-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 12
Registry:
    editor.focus.mode.color.light=5b595c
    ide.editor.tab.selection.animation=true
    debugger.watches.in.variables=false
    debugger.new.tool.window.layout=true
    editor.caret.width=1
    run.processes.with.pty=true
    vcs.new.widget=true
    ide.new.editor.tabs.vertical.borders=true
    ide.experimental.ui=true
    ide.balloon.shadow.size=0
    editor.focus.mode.color.dark=5b595c

Non-Bundled Plugins:
    de.femtopedia.diffplugin (2.1.10)
    intellij-awk (0.4.0)
    org.toml.lang (231.8109.91)
    com.nasller.CodeGlancePro (1.6.4)
    com.ultrahob.zerolength.plugin (0.95)
    org.elasticsearch4idea (2023.1.6-231)
    com.intellij.properties (231.8770.3)
    String Manipulation (9.7.1)
    ir.msdehghan.plugins.ansible (0.92)
    indent-rainbow.indent-rainbow (2.1.0)
    com.intellij.grazie.pro (0.3.224)
    com.kalessil.phpStorm.phpInspectionsEA (5.0.0.0)
    ru.adelf.idea.dotenv (2023.1)
    de.espend.idea.php.annotation (9.3.1)
    com.virtuslab.git-machete (4.0.0)
    izhangzhihao.rainbow.brackets (2023.2.8)
    pro.bashsupport (3.2.3.231)
    lv.midiana.misc.phpstorm-plugins.deep-keys (2022.08.03.001)
    fr.adrienbrault.idea.symfony2plugin (2022.1.250)
    com.chrisrm.idea.MaterialThemeUI (8.6.4)
    com.intellij.lang.jsgraphql (4.0.1)
    com.magento.idea.magento2plugin (5.1.1)
    org.sonarlint.idea (8.3.0.71062)
    com.mallowigi (90.0.0)
    zielu.gittoolbox (500.0.9+213)
    mobi.hsz.idea.gitignore (4.5.0)
    com.github.blarc.gitlab-template-lint-plugin (1.13.0)

Current Desktop: ubuntu:GNOME
```

**Comment by Blarc** [comment]
Hey @an2riy,

why do you think the re-indexation is triggered because of the Gitlab Template Lint plugin? Would love to have some more information to know how to tackle this problem.

Thanks.

**Comment by an2riy** [comment]
Hello.
When I disabled the plugin this hasn't happened.
**I have invalidated all caches and it seems that the problem disappeared.**
Only once appeared the exception:
```
Access is allowed from Event Dispatch Thread (EDT) only

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Current thread: Thread[DefaultDispatcher-worker-1,5,main] 1121942000 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 926683878
	at com.intellij.openapi.application.impl.ApplicationImpl.throwThreadAccessException(ApplicationImpl.java:1050)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertIsDispatchThread(ApplicationImpl.java:1037)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getSelectedUserObjects(AbstractProjectViewPane.java:346)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getData(AbstractProjectViewPane.java:400)
	at com.intellij.ide.projectView.impl.ProjectViewImpl$MyPanel.getData(ProjectViewImpl.java:1304)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProviderInner(DataManagerImpl.java:241)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProviderAndRulesInner(DataManagerImpl.java:81)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProviderAndRules(DataManagerImpl.java:60)
	at com.intellij.openapi.actionSystem.impl.EdtDataContext.calcData(EdtDataContext.java:162)
	at com.intellij.openapi.actionSystem.impl.EdtDataContext.getDataInner(EdtDataContext.java:147)
	at com.intellij.openapi.actionSystem.impl.EdtDataContext.getData(EdtDataContext.java:112)
	at com.intellij.openapi.actionSystem.DataContext.getData(DataContext.java:45)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener$execute$2$1.invoke(ApplicationStartupListener.kt:22)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener$execute$2$1.invoke(ApplicationStartupListener.kt:21)
	at com.intellij.openapi.application.ActionsKt.runReadAction$lambda$3(actions.kt:19)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:878)
	at com.intellij.openapi.application.ActionsKt.runReadAction(actions.kt:19)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener$execute$2.invoke(ApplicationStartupListener.kt:21)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener$execute$2.invoke(ApplicationStartupListener.kt:20)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener.execute$lambda$0(ApplicationStartupListener.kt:20)
	at org.jetbrains.concurrency.AsyncPromise$onSuccess$1.invoke(AsyncPromise.kt:89)
	at org.jetbrains.concurrency.AsyncPromise$onSuccess$1.invoke(AsyncPromise.kt:83)
	at org.jetbrains.concurrency.AsyncPromise.onSuccess$lambda$3(AsyncPromise.kt:83)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
	at org.jetbrains.concurrency.AsyncPromise.onSuccess(AsyncPromise.kt:83)
	at org.jetbrains.concurrency.AsyncPromise.onSuccess(AsyncPromise.kt:13)
	at com.github.blarc.gitlab.template.lint.plugin.listeners.ApplicationStartupListener.execute(ApplicationStartupListener.kt:20)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$4$1$1.invokeSuspend(StartupManagerImpl.kt:254)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$4$1$1.invoke(StartupManagerImpl.kt)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$4$1$1.invoke(StartupManagerImpl.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$4$1.invokeSuspend(StartupManagerImpl.kt:253)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

Thanks
